python/activator/activator.py

```diff
@@ -208,20 +262,19 @@ def next_visit_handler() -> Tuple[str, int]:
                 )
                 break
 
-            ack_list = []
-            for received in response.received_messages:
-                ack_list.append(received.ack_id)
-                oid = received.message.attributes["objectId"]
-                try:
-                    raw_info = Snap.from_oid(oid)
-                    _log.debug("Received %r", raw_info)
-                    if raw_info.is_consistent(expected_visit):
-                        # Ingest the snap
-                        mwi.ingest_image(oid)
-                        expid_set.add(raw_info.exp_id)
-                except ValueError:
-                    _log.error(f"Failed to match object id '{oid}'")
-            subscriber.acknowledge(subscription=subscription.name, ack_ids=ack_list)
+            for received in messages:
+                for oid in _parse_bucket_notifications(received.value()):
+                    try:
+                        raw_info = Snap.from_oid(oid)
+                        _log.debug("Received %r", raw_info)
+                        if raw_info.is_consistent(expected_visit):
+                            # Ingest the snap
+                            mwi.ingest_image(oid)
+                            expid_set.add(raw_info.exp_id)
+                    except ValueError:
+                        _log.error(f"Failed to match object id '{oid}'")
+            # Don't commit message, it causes other messages to be dropped as well.
```

Comment:
Which other messages get dropped?  At worst, I would think we can commit when we've processed everything that was returned by `consume()`.

Reply:
This was based on what I'd read in the [Kafka docs](https://docs.confluent.io/platform/current/clients/consumer.html) and [a comment on Slack](https://lsstc.slack.com/archives/C0345PUU4N9/p1666802736907339?thread_ts=1666740764.692819&cid=C0345PUU4N9) from @dspeck1. Maybe I misunderstood, but I thought Kafka can only commit ranges of messages, and that potentially meant messages for other detectors or (with enough mixing) even visits.

Reply:
The point to the Slack message was that a given _consumer group_ commits all at once.  Since every worker will have its own consumer group, it's not a problem to commit everything it has seen.